src/realm/sort_descriptor.cpp

```diff
@@ -478,7 +533,15 @@ void BaseDescriptor::Sorter::cache_first_column(IndexPairs& v)
         }
 
         const auto obj = col.table->get_object(key);
-        index.cached_value = ck.get_value(obj);
```

Comment:
`cache_first_column` is called multiple times on the same IndexPairs structure when there are multiple sort descriptors applied. So you need to make sure that the values are always overwritten rather than assuming that they have been default initialized.